src/components/Clasification.jsx

```diff
@@ -0,0 +1,106 @@
+import { useSelector, useDispatch } from 'react-redux';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { fetchPokemon, fetchType } from '../redux/pokemon/pokemonSlice';
+import { filterType } from './Pokemon';
+import pokemonImg from '../assets/pokemon.png';
+
+const Clasification = () => {
+  let typesFiltered;
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const [loadingType, setLoadingType] = useState(true);
+  const [loadingPokemon, setLoadingPokemon] = useState(true);
+  const [search, setSearch] = useState('');
+
+  const pokemon = useSelector(
+    (state) => state.pokemon.pokemon,
+  );
+
+  const types = useSelector(
+    (state) => state.pokemon.type.results,
+  );
+
+  useEffect(() => {
+    dispatch(fetchType('pokemon')).then(() => {
+      setLoadingType(false);
+    });
+  }, [dispatch]);
+
+  useEffect(() => {
+    dispatch(fetchPokemon('pokemon')).then(() => {
+      setLoadingPokemon(false);
+    });
+  }, [dispatch]);
```

Comment:
- [ ] Secondary, update this useEffect to this. :+1: :+1: 

```sh
  useEffect(() => {
    if (pokemon.length === 0) {
      setLoadingPokemon(true);
      dispatch(fetchPokemon('pokemon')).then(() => {
        setLoadingPokemon(false);
      });
    }
  }, [dispatch]);
```

Reply:
well resolved! ✔️